docs/admin/30-configuration/30-keycloak.md

```diff
@@ -1,6 +1,161 @@
 ---
 sidebar_position: 3
 id: keycloak
-title: Keycloak
-draft: true
----
\ No newline at end of file
+title: Keycloak Integration
+---
+
+# Keycloak Integration
+
+OpenCloud supports using Keycloak as an external identity provider, providing enterprise-grade identity management capabilities. This guide explains how to set up and configure Keycloak with OpenCloud.
+
+## Overview
+
+OpenCloud employs a dual-track authentication strategy:
+
+1. **Built-in IDP (Identity Provider)**: 
+   - Based on LibreGraph Connect (lico)
+   - Targeted at smaller installations (up to 500 users)
+   - Designed for standalone or small deployments
+
+2. **External Identity Providers**:
+   - Keycloak as the recommended OIDC provider for larger installations
+   - Support for Azure AD, EntraID, ADFS through Keycloak
+   - Enterprise-focused solution
+
+## Setting Up Keycloak with Docker Compose
+
+OpenCloud includes a pre-configured Keycloak setup in the Docker Compose examples. To enable Keycloak:
+
+1. Navigate to your OpenCloud deployment directory
+2. Edit the `.env` file and uncomment the Keycloak line:
+   ```
+   KEYCLOAK=:keycloak.yml
+   ```
+3. Configure the following environment variables in the same file:
+   ```
+   KEYCLOAK_DOMAIN=your-domain.example.com
+   KEYCLOAK_REALM=openCloud
+   KEYCLOAK_ADMIN_USER=admin
+   KEYCLOAK_ADMIN_PASSWORD=your-secure-password
+   ```
+4. Start the deployment with Keycloak:
+   ```bash
+   docker-compose -f docker-compose.yml -f keycloak.yml up -d
```

Comment:
```suggestion
   docker-compose up -d
```

Reply:
We do not need to provide the file paths. The magic lies in the last line of the `.env` file 🪄